terraform/azure/modules/3_applications/linkerd/helm.tf

```diff
@@ -0,0 +1,126 @@
+# Deploys linkerd, linkerd control plane and linkerd visualization dashboard 
+# ##############################################
+
+# provider "azurerm" {
+#   features {}
+# }
+
+data "azurerm_resource_group" "rg" {
+  name = var.resource_group_name 
+}
+
+data "azurerm_kubernetes_cluster" "aks_cluster" {
+  name                = var.aks_cluster_name
+  resource_group_name = var.resource_group_name
+}
+
+data "azurerm_kubernetes_cluster" "aks_cluster_auth" {
+  name                = var.aks_cluster_name
+  resource_group_name = var.resource_group_name
+}
+
+provider "helm" {
```

Comment:
Please remove provider fromthe module, it will be used at the root level when calling. Instead create a provider.tf file that has only required providers. See https://github.com/CDCgov/NEDSS-Infrastructure/tree/main/terraform/azure/modules/3_applications/fluentbit and its README